tests/libtest/lib1518.c

```diff
@@ -81,7 +81,7 @@ int test(char *URL)
   curl_easy_getinfo(curl, CURLINFO_REDIRECT_COUNT, &curlRedirectCount);
   curl_easy_getinfo(curl, CURLINFO_EFFECTIVE_URL, &effectiveUrl);
   curl_easy_getinfo(curl, CURLINFO_REDIRECT_URL, &redirectUrl);
-  res = curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writecb);
+  test_setopt(curl, CURLOPT_WRITEFUNCTION, writecb);
```

Comment:
I've noticed, not specific to this PR, that some of the tests use macro [test_setopt](https://github.com/curl/curl/blob/curl-8_3_0/tests/libtest/test.h#L49-L51) and some use macro [easy_setopt](https://github.com/curl/curl/blob/curl-8_3_0/tests/libtest/test.h#L173-L193). 

tests dir:
~~~
$ git grep '[^_]easy_setopt' |wc -l
    239

$ git grep test_setopt |wc -l
    676
~~~

What is the purpose of using macro test_setopt instead of macro easy_setopt? If the option can't be set then the former doesn't set an error message while the latter does. Both quit (goto test_cleanup). Wouldn't reliability be improved by always using the easy_setopt?

Reply:
I didn't notice that there were two macros and I thought I *was* using `easy_setopt`. `easy_setopt` is a bit broken, though, in that it will exit if a *previous* function set res to an error (`easy_setopt` does not always set `res`). These macros should probably be unified (and that problem fixed), but maybe there's a case where both need to exist.